pkg/vault/vault_test.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package vault
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	hashivault "github.com/hashicorp/vault/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadSecret(t *testing.T) {
+	mockKVStore := newMockKVStore()
+	mockKVStore.On("Get", mock.Anything, "test/secret1").Return(&hashivault.KVSecret{
+		Data: map[string]interface{}{
+			"value": "foo1",
+		},
+	}, nil)
+
+	mockKVStore.On("Get", mock.Anything, "test/secret2").Return(&hashivault.KVSecret{
+		Data: map[string]interface{}{
+			"value": "foo2",
+		},
+	}, nil)
+
+	mockKVStore.On("Get", mock.Anything, "test/secret3").Return(&hashivault.KVSecret{}, errors.New("vault error"))
+
+	mimirVaultClient := Vault{
+		KVStore: mockKVStore,
+	}
+
+	tests := map[string]struct {
+		path          string
+		expectError   bool
+		expectedValue string
+	}{
+		"read secret1 with no error": {
+			path:          "test/secret1",
+			expectError:   false,
+			expectedValue: "foo1",
+		},
+		"read secret2 with no error": {
+			path:          "test/secret2",
+			expectError:   false,
+			expectedValue: "foo2",
+		},
+		"read secret with non-existent path": {
+			path:        "test/secret3",
+			expectError: true,
+		},
+	}
+
+	for testName, testCase := range tests {
+		t.Run(testName, func(t *testing.T) {
+			secret, err := mimirVaultClient.ReadSecret(testCase.path)
+			if testCase.expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, testCase.expectedValue, string(secret))
+			}
+		})
+	}
+}
+
+type mockKVStore struct {
+	mock.Mock
```

Comment:
Instead of using `mock.Mock`, we can have `values map[string]interface{}` and return requested path from `Get` call.

Mocks are often fragile and tightly coupled to implementation.